src/fswAlgorithms/stateEstimation/thrustCMEstimation/thrustCMEstimation.h

```diff
@@ -0,0 +1,78 @@
+/*
+ ISC License
+ 
+ Copyright (c) 2023, Laboratory  for Atmospheric and Space Physics, University of Colorado at Boulder
+ 
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ 
+ */
+
+
+/*! @brief Top level structure for the thrust CM estimation kalman filter.
+ Used to estimate the spacecraft's center of mass position with respect to the B frame.
+ */
+
+#ifndef THRUSTCMESTIMATION_H
+#define THRUSTCMESTIMATION_H
+
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/macroDefinitions.h"
+#include "architecture/msgPayloadDefC/THRConfigMsgPayload.h"
+#include "architecture/msgPayloadDefC/CmdTorqueBodyMsgPayload.h"
+#include "architecture/msgPayloadDefC/AttGuidMsgPayload.h"
+#include "architecture/msgPayloadDefC/VehicleConfigMsgPayload.h"
+#include "cMsgCInterface/VehicleConfigMsg_C.h"
+#include "architecture/msgPayloadDefC/CMEstDataMsgPayload.h"
+
+#include <string.h>
+#include <array>
+#include <math.h>
+
+class ThrustCMEstimation: public SysModel {
+public:
+    ThrustCMEstimation();
+    ~ThrustCMEstimation() override;
+    void SelfInit();                   //!< Self initialization for C-wrapped messages
+    void Reset(uint64_t CurrentSimNanos) override;
+    void UpdateState(uint64_t CurrentSimNanos) override;
+
+    /*! declare these user-defined quantities */
+    double attitudeTol;
+
+    ReadFunctor<THRConfigMsgPayload>        thrusterConfigBInMsg;
+    ReadFunctor<CmdTorqueBodyMsgPayload>    intFeedbackTorqueInMsg;
+    ReadFunctor<AttGuidMsgPayload>          attGuidInMsg;
+    ReadFunctor<VehicleConfigMsgPayload>    vehConfigInMsg;
+    Message<CMEstDataMsgPayload>            cmEstDataOutMsg;
+    Message<VehicleConfigMsgPayload>        vehConfigOutMsg;
+    VehicleConfigMsg_C                      vehConfigOutMsgC = {};
+
+    Eigen::Vector3d r_CB_B;
+    Eigen::Vector3d P0;
+    Eigen::Vector3d R0;
```

Comment:
:small_red_triangle_down: I think these three member variables should be `private` and passed as parameters to the module's constructor. They are initial values which are used in reset. I think they shouldn't be settable though out a sim. Rather, if a user wants to change these they should create a new module. Happy to chat more about this and hear your thoughts.